### PR TITLE
Accessibility - Transition from interceptor to accessibility library

### PR DESCRIPTION
**Pull request by lm-n**
* added accessible elements to iframe using accessibility library
* removed interceptor
* added aria labels
* added css to iframe
* focus on iframe when play is pressed

@MathuraMG @CleezyITP

### CODE REVIEW

**Comment by catarak** [comment]
yay this is very exciting! in `README.md`, in the installation instructions, there's a section about the submodule setup—could you remove those parts in this PR as well? thanks!

File: client/modules/IDE/components/PreviewFrame.jsx [hunk 1]
```diff
@@ -198,49 +198,31 @@ class PreviewFrame extends React.Component {
...
-      if (this.props.soundOutput) {
-        let soundInterceptorScripts = [];
-        soundInterceptorScripts = [
-          '/p5-interceptor/soundInterceptor/interceptorP5.js'
-        ];
-        interceptorScripts = interceptorScripts.concat(soundInterceptorScripts);
-      }
-      scriptsToInject = scriptsToInject.concat(interceptorScripts);
+    const accessiblelib = sketchDoc.createElement('script');
+    accessiblelib.setAttribute('src', 'https://cdn.rawgit.com/MathuraMG/p5-accessibility/e856365c/dist/p5-accessibility.js');
```
Comment (catarak) [inline client/modules/IDE/components/PreviewFrame.jsx]: rather than link to the accessibility using raw git, i think it would be better to add the library as a dependency in `package.json`, then import the built script using webpack.

File: client/modules/IDE/reducers/files.js [hunk 2]
```diff
@@ -13,15 +13,19 @@ const defaultHTML =
 `<!DOCTYPE html>
 <html>
   <head>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
```
Comment (catarak) [inline client/modules/IDE/reducers/files.js]: why is this including jquery?

File: client/modules/IDE/components/PreviewFrame.jsx [hunk 3]
```diff
@@ -419,7 +401,7 @@ PreviewFrame.propTypes = {
   textOutput: PropTypes.bool.isRequired,
   gridOutput: PropTypes.bool.isRequired,
   soundOutput: PropTypes.bool.isRequired,
-  setTextOutput: PropTypes.func.isRequired,
+  // setTextOutput: PropTypes.func.isRequired,
```
Comment (catarak) [inline client/modules/IDE/components/PreviewFrame.jsx]: should this be removed rather than commented out?

**Comment by MathuraMG** [comment]
@catarak - We will fix the above asap. 
Also, a quick update. There is an issue with the accessibility library that we found yesterday. We will fix that and then get back on this PR, so that we have the right version of the library put in. Is that ok?
Thanks!

@lm-n

**Comment by catarak** [comment]
no worries, that is of course okay! this brings up an important point, which is how we want to connect the version of the accessibility library to the editor in the `package.json`, so that when you run `npm install` in the editor, it doesn't install the latest commit, but a stable commit— we could 1. add a commit sha, or 2. tag a release version, using [semantic versioning](https://semver.org/). (1) could come first and then (2) could come later.

File: client/modules/IDE/reducers/files.js (hunk 2 above)
Comment (MathuraMG) [reply client/modules/IDE/reducers/files.js]: we will clear this out - our library has dependencies on jquery but we will make them vanilla javascript.

File: client/modules/IDE/components/PreviewFrame.jsx (hunk 3 above)
Comment (MathuraMG) [reply client/modules/IDE/components/PreviewFrame.jsx]: Will do.

File: client/modules/IDE/components/PreviewFrame.jsx (hunk 1 above)
Comment (MathuraMG) [reply client/modules/IDE/components/PreviewFrame.jsx]: cc @lm-n, @cleezyITP - We have been talking about this one, we would like the library to be a part of the HTML code as a CDN so that when a user shares their code (say, send a zip to someone, it will include the library. This would make sharing more accessible. What do you think?

Comment (MathuraMG) [reply client/modules/IDE/components/PreviewFrame.jsx]: @catarak - was wondering what your thoughts on this are?

File: client/modules/IDE/reducers/files.js [hunk 4]
```diff
@@ -18,10 +18,13 @@ const defaultHTML =
     <script src="https://cdnjs.cloudflare.com/ajax/libs/p5.js/0.5.11/addons/p5.sound.min.js"></script>
     <link rel="stylesheet" type="text/css" href="style.css">
     <meta charset="utf-8" />
+
   </head>
   <body>
+    <section aria-label='accessible output' id='accessible-outputs'></section>
```
Comment (catarak) [inline client/modules/IDE/reducers/files.js]: rather than add the accessibility elements and css to the default html file, i think it would be better to inject them in the `PreviewFrame.jsx` file, in the function `injectLocalFiles`. As it is now, a user could accidentally delete these elements from the HTML and then the accessibility features would be broken.

File: client/modules/IDE/components/PreviewFrame.jsx [hunk 5]
```diff
@@ -198,49 +198,31 @@ class PreviewFrame extends React.Component {
...
-      if (this.props.soundOutput) {
-        let soundInterceptorScripts = [];
-        soundInterceptorScripts = [
-          '/p5-interceptor/soundInterceptor/interceptorP5.js'
-        ];
-        interceptorScripts = interceptorScripts.concat(soundInterceptorScripts);
-      }
-      scriptsToInject = scriptsToInject.concat(interceptorScripts);
+    const accessiblelib = sketchDoc.createElement('script');
+    accessiblelib.setAttribute('src', 'https://cdn.rawgit.com/MathuraMG/p5-accessibility/9cb5bd0b/dist/p5-accessibility.js');
```
Comment (catarak) [inline client/modules/IDE/components/PreviewFrame.jsx]: i still would prefer to link this file by putting it in the package.json and importing the build with webpack!

File: client/modules/IDE/reducers/files.js (hunk 4 above)
Comment (MathuraMG) [reply client/modules/IDE/reducers/files.js]: @catarak - I think some of the questions we had for you were deleted after we made the new commits :) 
@lm-n and myself were thinking it should be a part of the HTML (at the risk of being deleted) , the same way as p5-dom or p5-sound. This way if someone decides to share their code, they can share it with the accessibility lib and so the code stays accessible beyond the editor. What do you think about this?

Comment (catarak) [reply client/modules/IDE/reducers/files.js]: got it. should it be added by default or only if accessibility features are turned on? and also, you want the script to point to a specific release of the library? or a specific commit? or to the latest on master?

Comment (MathuraMG) [reply client/modules/IDE/reducers/files.js]: I think we tried to add it only if accessibility features are added but there was a problem? @lm-n ?

**Comment by lm-n** [comment]
@catarak both should work. When I move into the accessibility branch the submodule directory does not exist anymore. It can be done also with `git submodule deinit -f [submodule name]` `rm -rf [submodule directory]` `git rm -f [submodule name]`

File: client/modules/IDE/reducers/files.js [hunk 6]
```diff
@@ -30,6 +31,15 @@ const defaultCSS =
   margin: 0;
   padding: 0;
 }
+
+#accessible-outputs {
```
Comment (catarak) [inline client/modules/IDE/reducers/files.js]: i think this is the last thing, but this CSS should also get injected by the library!

File: client/modules/IDE/components/PreviewFrame.jsx (hunk 5 above)
Comment (catarak) [reply client/modules/IDE/components/PreviewFrame.jsx]: @MathuraMG and i talked through this and decided it would be best to address this in the future ✨

File: client/modules/IDE/components/PreviewFrame.jsx [hunk 7]
```diff
@@ -207,6 +207,12 @@ class PreviewFrame extends React.Component {
     accessiblelib.setAttribute('src', 'https://cdn.rawgit.com/MathuraMG/p5-accessibility/9cb5bd0b/dist/p5-accessibility.js');
     const accessibleOutputs = sketchDoc.createElement('section');
```
Comment (catarak) [inline client/modules/IDE/components/PreviewFrame.jsx]: should these accessible outputs be created in the library itself? this is necessary for the library to work, right?

Comment (MathuraMG) [reply client/modules/IDE/components/PreviewFrame.jsx]: The reason we don't create the sections in the library is because the section is [currently the trigger](https://github.com/mathuramg/p5-accessibility#usage) to which output the user wants. So in this case we create that section when they select the option in preferences.

@lm-n

Comment (catarak) [reply client/modules/IDE/components/PreviewFrame.jsx]: got it. would it instead make sense to make some boolean for the library instead that enables/disables the accessible output? then, in the future, if this ever needs to change you can change it in the accessibility library rather than here as well.

Comment (MathuraMG) [reply client/modules/IDE/components/PreviewFrame.jsx]: Sorry, I didn't get what you mean by ' boolean for the library'? :)
The reason we are having sections now is so that beyond the editor the user can determine where they want the output in their program.

Comment (catarak) [reply client/modules/IDE/components/PreviewFrame.jsx]: got it! what i mean is maybe there's a script that initializes the library, rather than just appending to an HTML element if it's there. for example, one includes the accessibility library, and then includes a script:
```JS
var accessibility = new p5.Accessibility(<reference to the HTML element you want to append to>);
```
or it could create an HTML element to append to by default. My inspiration for this is [dat.gui](https://github.com/dataarts/dat.gui). my instinct is that it is not the best that including an html element determines whether or not the library is included.

Comment (MathuraMG) [reply client/modules/IDE/components/PreviewFrame.jsx]: Got it!  Yes, we will look more into this. 👍 
Would be okay if we track these refactors in a different issue that we tackle following this?

Comment (catarak) [reply client/modules/IDE/components/PreviewFrame.jsx]: sure! should i merge this in then?

Comment (lm-n) [reply client/modules/IDE/components/PreviewFrame.jsx]: sounds good!

Comment (lm-n) [reply client/modules/IDE/components/PreviewFrame.jsx]: Here are the issues on the library repo: https://github.com/MathuraMG/p5-accessibility/issues/13 https://github.com/MathuraMG/p5-accessibility/issues/12

Comment (MathuraMG) [reply client/modules/IDE/components/PreviewFrame.jsx]: @catarak Yes! That sounds good!

Comment (catarak) [reply client/modules/IDE/components/PreviewFrame.jsx]: great, will do ⭐️

**Comment by catarak** [comment]
just another thought that there's an open issue for, #137: right now, accessibility features won't be added to the full screen view or embed view. and if a user downloads a sketch it won't be included. open to thoughts!
